CHANGELOG.md

```diff
@@ -1,1929 +1,267 @@
+  [`6800f76ff`](https://github.com/npm/npm/commit/6800f76ffcd674742ba8944f11f6b0aa55f4b612)
+  [`ec90c06c7`](https://github.com/npm/npm/commit/ec90c06c78134eb2618612ac72288054825ea941)
+  [`825b5d2c6`](https://github.com/npm/npm/commit/825b5d2c60e620da5459d9dc13d4f911294a7ec2)
+  [`4785f13fb`](https://github.com/npm/npm/commit/4785f13fb69f33a8c624ecc8a2be5c5d0d7c94fc)
+  [`bd16485f5`](https://github.com/npm/npm/commit/bd16485f5b3087625e13773f7251d66547d6807d)
+  Restore the ability to bundle dependencies that are uninstallable from the
+  registry.  This also eliminates needless registry lookups for bundled
+  dependencies.
 
-### BUG FIXES TO TOKENS AND PROFILES
-
-* [`a66e0cd03`](https://github.com/npm/npm/commit/a66e0cd0314893b745e6b9f6ca1708019b1d7aa3)
-  For CIDR filtered tokens, allow comma separated CIDR ranges, as documented. Previously
-  you could only pass in multiple cidr ranges with multiple `--cidr` command line options.
-  ([@iarna](https://github.com/iarna))
-* [`d259ab014`](https://github.com/npm/npm/commit/d259ab014748634a89cad5b20eb7a40f3223c0d5)
-  Fix token revocation when an OTP is required.  Previously you had to pass
-  it in via `--otp`.  Now it will prompt you for an OTP like other
-  `npm token` commands.
-  ([@iarna](https://github.com/iarna))
-* [`f8b1f6aec`](https://github.com/npm/npm/commit/f8b1f6aecadd3b9953c2b8b05d15f3a9cff67cfd)
-  Update token and profile commands to support legacy (username/password) authentication.
-  (The npm registry uses tokens, not username/password pairs, to authenticate commands.)
+  Fixed a bug where attempting to install a dependency that is bundled
+  inside another module without reinstalling that module would result in
+  ENOENT errors.
   ([@iarna](https://github.com/iarna))
+* [`429498a8c`](https://github.com/npm/npm/commit/429498a8c8d4414bf242be6a3f3a08f9a2adcdf9)
+  [#20029](https://github.com/npm/npm/pull/20029)
+  Allow packages with non-registry specifiers to follow the fast path that
+  the we use with the lock-file for registry specifiers. This will improve install time
+  especially when operating only on the package-lock (`--package-lock-only`).
 
-### OTHER BUG FIXES
+  ([@zkat](https://github.com/zkat))
 
-* [`6954dfc19`](https://github.com/npm/npm/commit/6954dfc192f88ac263f1fcc66cf820a21f4379f1)
-  Fix a bug where packages would get pushed deeper into the tree when upgrading without
-  an existing copy on disk. Having packages deeper in the tree ordinarily is harmless but
-  is not when peerDependencies are in play.
-  ([@iarna](https://github.com/iarna))
-* [`1ca916a1e`](https://github.com/npm/npm/commit/1ca916a1e9cf94691d1ff2e5e9d0204cfaba39e1)
-  Fix bug where when switching from a linked module to a non-linked module, the dependencies
-  of the module wouldn't be installed on the first run of `npm install`.
+  Fix the a bug where `npm i --only=prod` could remove development
```

Comment:
Missing `*` and shasum?

Reply:
Not as such… this was bug fix was drawn out of the other through a rather complicated series of circumstances. (The change to fakeChildren handling resulted in tests failing because of the `--only=prod` bug now showing because their children were now being inflated as fakeChildren. However the `--only=prod` bug as active for ANY dependency instantiated as a fakeChild. We just didn't have tests that hit that code path, due to most of the tests using `file:` dependencies.

Reply:
Arguably it could have been a separate commit. I ended up squashing it into the original as I was merging.